mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DaoBase.java

```diff
@@ -60,6 +60,11 @@
 
   protected static CompletionStage<PreparedStatement> prepare(
       SimpleStatement statement, MapperContext context) {
+    if (context.getExecutionProfileName() != null) {
+      statement = statement.setExecutionProfileName(context.getExecutionProfileName());
+    } else if (context.getExecutionProfile() != null) {
+      statement = statement.setExecutionProfile(context.getExecutionProfile());
+    }
     return context.getSession().prepareAsync(statement);
```

Comment:
Do we prevent users from setting both `profileName` and profile (`class`)?
If no, did we document the fact that `getExecutionProfileName` has priority over `getExecutionProfile`?

Reply:
It's enforced implicitly by the mapper processor: only one `@DaoProfile`-annotated parameter is allowed per DAO factory method.

I'll add a comment on the context methods to explain that.

Reply:
Now that I think of it, we could also allow injection of the profile via `MapperBuilder`, like we do with the default keyspace. I'll add that.